pkgs/servers/home-assistant/component-packages.nix

```diff
@@ -0,0 +1,428 @@
+# Generated from parse-requirements.py
+# Do not edit!
+
+{
+  "nuimo_controller" = ps: with ps; [  ];
+  "bbb_gpio" = ps: with ps; [  ];
+  "doorbird" = ps: with ps; [  ];
+  "isy994" = ps: with ps; [  ];
+  "notify.html5" = ps: with ps; [  ];
+  "sensor.mvglive" = ps: with ps; [  ];
+  "arduino" = ps: with ps; [  ];
+  "xiaomi_aqara" = ps: with ps; [  ];
+  "rpi_gpio" = ps: with ps; [  ];
+  "remember_the_milk" = ps: with ps; [ httplib2 ];
```

Comment:
Are these all normalized names? https://www.python.org/dev/peps/pep-0503/#normalized-names

Reply:
The names on the left are component names, no need to normalize them.
However, normalization is indeed important for finding the appropriate package. Is it the case that all `pname`s inside Nixpkgs should be normalized?

Reply:
I wish we could use normalized `pname` but we can't unfortunately because of https://github.com/pypa/warehouse/issues/2537

Reply:
Okay, then I don't have to normalize anything. Howver, a case-insensitive search seems to find some more packages. So, I use `re.I` now.

Reply:
Attributes on the other hand should preferably be normalized names.

Reply:
The attributes need to be called exactly as the components in order to generate a list of used components from the config. Otherwise I would have to write a function in Nix that normalizes names and then I'd have to recursively list all attribute paths of the config, normalize them and check if they appear in `comonent-packages.nix`.